infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/SessionManager.java

```diff
@@ -0,0 +1,362 @@
+                        }
+
+                        sessionToContextMap.put(session, context);
+                    }
+                }
+            }
+            txn.commit();
+        }
+
+        log.info("Total sessions={}, outgoing={}, incoming={}, sessions={}", sessions.size(), outgoingSessions.size(),
+                incomingSessions.size(), sessions);
+    }
+
+    /**
+     * Retrieve all sessions for which this cluster is the sink
+     *
+     * @return set of sessions for which this cluster is the sink
+     */
+    public Set<LogReplicationSession> getIncomingSessions() {
+        return incomingSessions;
+    }
+
+    /**
+     * Retrieve all sessions from the given source cluster
+     *
+     * @param sourceClusterId   the identifier of the source cluster
+     * @return set of sessions for which this cluster is the sink
+     */
+    public Set<LogReplicationSession> getIncomingSessions(String sourceClusterId) {
+        return incomingSessions.stream()
+                .filter(s -> s.getSinkClusterId().equals(sourceClusterId)).collect(Collectors.toSet());
+    }
+
+    /**
+     * Retrieve all sessions for which this cluster is the source
+     *
+     * @return set of sessions for which this cluster is the sink
+     */
+    public Set<LogReplicationSession> getOutgoingSessions() {
+        return outgoingSessions;
+    }
+
+    /**
+     * Retrieve all sessions to the given sink cluster
+     *
+     * @param sinkClusterId   the identifier of the sink cluster
+     * @return set of sessions for which this cluster is the sink
+     */
+    public Set<LogReplicationSession> getOutgoingSessions(String sinkClusterId) {
+        return outgoingSessions.stream()
+                .filter(s -> s.getSinkClusterId().equals(sinkClusterId)).collect(Collectors.toSet());
+    }
+
+    /**
+     * Reset replication status for all sessions
+     */
+    public void resetReplicationStatus() {
+        metadataManager.resetReplicationStatus();
+    }
+
+    /**
+     * Retrieve replication status for all sessions
+     *
+     * @return  map of replication status per session
+     */
+    public Map<LogReplicationSession, ReplicationStatus> getReplicationStatus() {
+        return metadataManager.getReplicationStatus();
+    }
+
+    public static ReplicationSubscriber getDefaultSubscriber() {
+        return ReplicationSubscriber.newBuilder()
+                .setClientId(UuidMsg.newBuilder()
+                        .setMsb(DEFAULT_CLIENT_ID.getMostSignificantBits())
+                        .setLsb(DEFAULT_CLIENT_ID.getMostSignificantBits())
+                        .build())
+                .setClientName(DEFAULT_CLIENT)
+                .setModel(ReplicationModel.FULL_TABLE)
+                .build();
+    }
+
+    public void stopReplication() {
+        log.info("Stopping log replication.");
+        replicationManager.stop();
+    }
+
+    public void stopReplication(Set<LogReplicationSession> sessions, boolean remove) {
```

Comment:
the boolean "remove" is never false currently. Is there somecase for which the boolean can be false?

Reply:
no, let me remove it